test/integration/test/recipe_test.js

```diff
@@ -0,0 +1,188 @@
+'use strict';
+
+const chakram = require('chakram'), expect = chakram.expect;
+var Prepper = require('../lib/testPrepper.js');
+
+Prepper.makeSuite('Add/Remove recipe',function(){
+
+  var source = {'name': 'Hellooo, Thinkerbell', 'rules':
+            [{'conditions':
+              [{'source':
+                [{'id':'OpenZWave72057594126794752 (Sensor)'}],
+              'kind':'OpenClosed',
+              'range':{'Eq':{'OpenClosed':'Open'}}
+            }],
+            'execute':
+              [{'destination':
+                [{'kind':'Log'}],
+                'kind':'Log',
+                'value':{'String':'Closed'}}]}
+              ]};
+    source = JSON.stringify(source);
+
+  before(Prepper.foxboxManager.foxboxLogin);
+
+  it('Check service list for recipe',function(){        
+    return chakram.get(Prepper.foxboxManager.serviceListURL)
+    .then(function(listResponse) {
+      expect(listResponse).to.have.status(200);
+      var found = false;
+      for (var entry of listResponse.body) {
+        if (entry.adapter === 'thinkerbell@link.mozilla.org') {
+          found = true;                
+        }
+      }
+      expect(found).equals(true);
+    });
+  });
+  
+  it('Add recipe',function(){
+    
+
+    var payload = {
+      'select':{
+         'kind':'AddThinkerbellRule'
+      },
+      'value':{
+         'ThinkerbellRule':{
+            'name':'First Recipe',
+            'source': source
+          }
+      }
+    };
+    return chakram.put(Prepper.foxboxManager.setterURL, payload)
+    .then(function(cmdResponse) {
+      expect(cmdResponse).to.have.status(200);
+      expect(cmdResponse.body['thinkerbell-add-rule']).equals(null);
+      return chakram.get(Prepper.foxboxManager.serviceListURL);
+    })
+    .then(function(listResponse) {
+      var found = false;
+      expect(listResponse).to.have.status(200);
+      for (var entry of listResponse.body) {
+        if (entry.adapter === 'thinkerbell@link.mozilla.org') {
+          if (entry.id === 'thinkerbell/First Recipe') {
+            found = true;
+          }
+        }
+      }
+      expect(found).equals(true);     
+    });
+  }); 
+
+  it('Add two recipes at the same time',function(){        
+    
+    var payload = {
+      'select':{
+         'kind':'AddThinkerbellRule'
+      },
+      'value':{
+         'ThinkerbellRule':{
+            'name':'Second Recipe',
```

Comment:
I thought about it, but I wanted to test a scenario where I add another recipe when a recipe is already present, and if I load a recipe in before(), the test would get too long.   Would above be allowed in such case?